cmake: Add zephyr base address when looking file into soc/

erwango opened the pull request:
When checking availability of soc/${ARCH}/CMakeLists.txt,
we should use ${ZEPHYR_BASE} as path base.

Fixes #9956

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

carlescufi reviewed (approved):
LGTM

erwango commented on CMakeLists.txt:
```diff
@@ -336,7 +336,7 @@ add_subdirectory(misc)
 include(misc/generated/CMakeLists.txt)
 
-if(EXISTS soc/${ARCH}/CMakeLists.txt)
+if(EXISTS ${ZEPHYR_BASE}/soc/${ARCH}/CMakeLists.txt)
```
I'm actually not sure this is correct.
This worked w/o support of out of tree SoC definition.
With support of out of tree SoC, I think ${SOC_DIR} would be better than ${ZEPHYR_BASE}, but I I'm confused with the syntax: `add_subdirectory(dir1 dir2)` does it add dir1 and dir2 as subdirs?

In that case shouldn't we test CMakeLists.txt availability in dir1 and dir2 and then split the whole section into in tree and out of tree tests?

nashif replied:
should actually be
```
if(EXISTS ${SOC_DIR}/${ARCH}/CMakeLists.txt)
```

erwango replied:
ok, makes sense. Updated. Txs

erwango committed:
cmake: Add zephyr base address when looking file into soc/

When checking availability of soc/${ARCH}/CMakeLists.txt,
we should use ${ZEPHYR_BASE} as path base.

Fixes #9956

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
